### PR TITLE
Implement ShellyPlus1PM

### PR DESCRIPTION
**Pull request by Sn0w3y**
Example JSON:

`{"ble":{},"cloud":{"connected":true},"input:0":{"id":0,"state":false},"mqtt":{"connected":false},"switch:0":{"id":0, "source":"init", "output":false, "apower":0.0, "voltage":231.3, "current":0.000, "aenergy":{"total":8629.000,"by_minute":[0.000,0.000,0.000],"minute_ts":1708858380},"temperature":{"tC":44.3, "tF":111.7}},"sys":{"mac":"80646FE34998","restart_required":false,"time":"11:53","unixtime":1708858386,"uptime":150390,"ram_size":260364,"ram_free":115308,"fs_size":458752,"fs_free":143360,"cfg_rev":22,"kvs_rev":2,"schedule_rev":0,"webhook_rev":0,"available_updates":{},"reset_reason":3},"wifi":{"sta_ip":"192.168.178.169","status":"got ip","ssid":"heizung","rssi":-48,"ap_client_count":0},"ws":{"connected":false}}`

### CODE REVIEW

File: io.openems.edge.io.shelly/src/io/openems/edge/io/shelly/shellyplus1pm/IoShellyPlus1PmImpl.java [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+		}
+	}
+
+	private void processHttpResult(JsonElement result, Throwable error) {
+		this._setSlaveCommunicationFailed(result == null);
+
+		Integer power = null;
+		Integer voltage = null;
+		Integer current = null;
+		boolean relay0 = false;
```
Comment (sfeilmeier) [inline io.openems.edge.io.shelly/src/io/openems/edge/io/shelly/shellyplus1pm/IoShellyPlus1PmImpl.java:151]: Sollte `Boolean relay0 = null` sein.

**Review by sfeilmeier — CHANGES_REQUESTED** [review]
Bitte nochmal überarbeiten 👍

**Comment by Sn0w3y** [comment]
@sfeilmeier ist erledigt, danke für den Hinweis ! :)

File: io.openems.edge.io.shelly/src/io/openems/edge/io/shelly/shellyplus1pm/IoShellyPlus1PmImpl.java [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+						this._setVoltageL3(voltage);
+						this._setCurrentL3(current);
+						break;
+					}
+				}
+			} catch (OpenemsNamedException e) {
+				this.logDebug(this.log, e.getMessage());
+			}
+
+			this._setRelay(relay0);
```
Comment (sfeilmeier) [inline io.openems.edge.io.shelly/src/io/openems/edge/io/shelly/shellyplus1pm/IoShellyPlus1PmImpl.java]: This must be outside the brackets.

File: io.openems.edge.io.shelly/test/io/openems/edge/io/shelly/shellyplus1pm/IoShellyPlus1PmImplTest.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+public class IoShellyPlus1PmImplTest {
+
+	private static final String COMPONENT_ID = "io0";
+
+	@Test
+	public void test() throws Exception {
+		new ComponentTest(new IoShellyPlus1PmImpl()) //
+				.addReference("httpBridgeFactory", DummyBridgeHttpFactory.ofDummyBridge()) //
+				.activate(MyConfig.create() //
```
Comment (sfeilmeier) [inline io.openems.edge.io.shelly/test/io/openems/edge/io/shelly/shellyplus1pm/IoShellyPlus1PmImplTest.java:39]: Please provide full JUnit test, including parsing the JSON response

**Review by sfeilmeier — APPROVED** [review]

**Pull request merged**